File: 1313.py
class Solution(object):
    def decompressRLElist(self, nums):
        freq = 0
        arr = []
        for i, v in enumerate(nums):
            if i % 2 == 0:
                freq = nums[i]
            else:
                arr.extend(([nums[i]]*freq))
        return arr


a = Solution()
print(a.decompressRLElist([1, 1, 2, 3]))

File: 1313.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var decompressRLElist = function (nums) {
    let arr = [];
    let freq = 0;
    for (var i = 0; i < nums.length; i++) {
        if (i % 2 == 0) {
            freq = nums[i];
        } else {
            arr.push(...Array(freq).fill(nums[i]));
        }
    }
    return arr;
};

console.log(decompressRLElist([1, 1, 2, 3]));
